### PR TITLE
Options docs, defaults and valid values

### LINKED ISSUE #583: Options documentation and values
Currently the documentation for the options is done in the default options file. This documentation should be moved out of this file and added separately to readthedocs.

Also, the options should raise an error if a wrong value is given. For example, the options for `table_type` are `acc`, `runtime`, `compare` and `local_min` thus if the option was set, by accident or otherwise, to `chi_sq` then the options class should raise an error before the fitting is done and not in the result processing.

### PR DESCRIPTION
**Pull request by wathen**
#### Description of Work

Fixes #583 

Removes `default_options.ini` and moves the documentation into the read the docs and the default values to the options class. Also, adds `valid` values for certain options.

Docs are [here](https://fitbenchmarking.readthedocs.io/en/583_options_docs/)

#### Testing Instructions

1. Check new docs are complete with respect to the old default options file
2. Check docs for adding new options 
3. Check that there are not references to `default_options.ini` 

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

File: docs/source/extending/options_extend.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+The options file must be a `.ini` formatted file
+(`see here <https://docs.python.org/3/library/configparser.html#supported-ini-file-structure>`__).
+
+To add a new option to one of the four sections ``MINIMIZERS``,
+``FITTING``, ``PLOTTING`` and ``LOGGING``, follow steps:
+
+1. Add new option to the corresponding ``DEFAULTS`` dictionary.
+2. **OPTIONAL** add accepted option values to the ``VALID`` dictionary.
+   options class.
```
[the remark this answers is not in the document]
Comment (wathen) [reply docs/source/extending/options_extend.rst]: Removed

File: docs/source/users/jacobian.rst [hunk 2]
```diff
@@ -4,10 +4,12 @@
 Jacobian options
 ################
 
-Currently FitBenchmarking uses SciPy's `numerical finite difference <https://github.com/scipy/scipy/blob/912c54cd6473225c902377df410258839511b863/scipy/optimize/_numdiff.py#L198>`_ approximation to the Jacobian. The finite difference schemes implemented are:
+Currently FitBenchmarking uses SciPy's `numerical finite difference <https://github.com/scipy/scipy/blob/912c54cd6473225c902377df410258839511b863/scipy/optimize/_numdiff.py#L198>`_ approximation to the Jacobian and uses analytical Jacobians for the CUTEst data set.
```
[the remark this answers is not in the document]
Comment (wathen) [reply docs/source/users/jacobian.rst:7]: As discussed, we will merge this into master and then backport as is. After the backport we will remove analytical from the release

File: docs/source/users/options/fitting_option.rst [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+    [FITTING]
+    use_errors: yes
+
+
+Jacobian method (:code:`jac_method`)
+------------------------------------
+
+This sets the options type of Jacobian used. Current Jacobian methods are:
+
+* ``anaylytic`` - This uses the analytic Jacobian extracted from the fitting problem.
```
[the remark this answers is not in the document]
Comment (wathen) [reply docs/source/users/options/fitting_option.rst:91]: See above

File: docs/source/users/options/fitting_option.rst [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+* ``SciPyFD`` - denotes the use of SciPy's finite difference Jacobian approximations
+
+**Note:** Currently analytic Jacobians are available are only available for
+problems that use the cutest parser, which are currently in the directories
+'SIF' and 'DIAMOND_SIF'.# default 'SciPyFD'
```
[the remark this answers is not in the document]
Comment (wathen) [reply docs/source/users/options/fitting_option.rst]: Fixed

File: docs/source/extending/options_extend.rst [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+3. Add tests in the following way:
+
+    - Each of the sections has it's own test file, for example, ``test_option_fitting`` have tests for the ``FITTING`` section.
+
+    - Add default tests to class called ``{Section}OptionTests``.
+
+    - Add user defined tests to class called ``User{Section}OptionTests``. These
+      should check that the user added option is set correctly and that if
+      there are accepted options an ``OptionsError`` is raised.
```
[the remark this answers is not in the document]
Comment (wathen) [reply docs/source/extending/options_extend.rst:27]: Done

File: fitbenchmarking/utils/options.py [hunk 6]
```diff
@@ -26,66 +93,88 @@ def __init__(self, file_name=None):
...
+            if isinstance(value, list):
+                set1 = set(value)
+                set2 = set(self.VALID[section][option])
+                value_check = set1.issubset(set2)
+            else:
+                value_check = value in self.VALID[section][option]
+            if not value_check:
+                self.error_message.append(
+                    "Value set in the options, {0}, is not in the VALID "
+                    "list {1}".format(value, self.VALID[section][option]))
```
[the remark this answers is not in the document]
Comment (wathen) [reply fitbenchmarking/utils/options.py]: Done

File: docs/source/users/options/fitting_option.rst (hunk 3 above)
Comment (tyronerees) [inline docs/source/users/options/fitting_option.rst:91]:
```suggestion
* ``analytic`` - This uses the analytic Jacobian extracted from the fitting problem.
```

**Comment by wathen** [comment]
> I spotted one small typo, otherwise looks good.
> 
> Shall I write an issue for checking the keys int he ini file are supported?

I fixed that actually and added tests to check that the 4 sections are set correctly as well as the keys in the ini file

**Pull request merged**